plugin-tester-java/src/main/java/example/myapp/helloworld/LoggingErrorHandlingGreeterServer.java

```diff
@@ -100,7 +111,19 @@ private static <ServiceImpl> Route loggingErrorHandlingGrpcRoute(
   //#custom-error-mapping
   private final static Function<Throwable, Trailers> customErrorMapping = (throwable) -> {
     if (throwable instanceof IllegalArgumentException) {
-      return new Trailers(Status.INVALID_ARGUMENT);
+      if (throwable.getMessage().equals(nameNonEmptyException.getMessage())) {
+        // We can pass through the message by attaching it to the Status.
+        return new Trailers(Status.INVALID_ARGUMENT.withDescription(throwable.getMessage()));
+      } else if (throwable.getMessage().equals(nameCapitalizedException.getMessage())) {
+        // We can pass through extra error details like localized versions of the message.
+        // FIXME: This subproject uses Java protos, which do not extend GeneratedMessage, so the code below does not compile.
```

Comment:
There's a bit of an issue here @johanandren 

Should we really be using scalapb.GeneratedMessage in the Java APIs? If so, then we need to either make the Scala LocalizedMessage available, or we need to find a GeneratedMessage that is appropriate to use here.

Reply:
Yes, good catch, for Java we should indeed have APIs for the Java proto message classes (`com.google.protobuf.Message`) instead of the ScalaPB ones. 

Looks like an oversight already in the pre-existing `GrpcServiceException.create`. I've PR:ed a fix for that in https://github.com/akka/akka-grpc/pull/1937 once that is merged you can rebase this on top and change the Java Trailers.create to accept a list of the Java type instead.

Reply:
Nice! Fixed in https://github.com/akka/akka-grpc/pull/1933/commits/f452a6a76d2667bb872c4c76f2ca54c8686bace3